vispy/gloo/context.py

```diff
@@ -160,7 +161,13 @@ class GLShared(object):
     # left, things should Just Work. 
     
     def __init__(self):
-        self._parser = GlirParser()
+        glir_file = config['glir_file']
+
+        parser_cls = GlirParser
+        if glir_file:
+            parser_cls = glir_logger(parser_cls, glir_file)
+
+        self._parser = parser_cls()
```

Comment:
Note the `parser.setter` a few lines below. I think these additions should go in there.